### ISSUE TITLE
Nullability: Failed pattern match makes compiler think variable is nullable

### ISSUE THREAD

**Issue by praeclarum** [dotnet/roslyn#31803]
In the following code:

```csharp
class Shape { public void Draw() {} }
class Circle : Shape {}

Shape shape = CreateShape ();
if (shape is Circle c) { }
else {
  shape.Draw(); // CS8602
}
```

The warning is raised even though the variable `shape` was declared non-nullable.

**Version Used**: .NET Core 3.0.100-preview-009812

**Expected Behavior**: Compiles OK.

**Actual Behavior**: Error CS8602

**Comment by Pilchie** [comment]
Hitting this in https://github.com/dotnet/project-system too.

**Comment by cston** [comment]
@praeclarum, thanks for reporting this issue.

This issue has been fixed in https://github.com/dotnet/roslyn/pull/30913.
